[AIRFLOW-5825] SageMakerEndpointOperator is not idempotent

OmairK opened the pull request:
- Added try except statement to check for duplicate endpoint while executing a create_endpoint request  which if found the request changes to update_endpoint
	- Added test case to test mock the duplicate endpoint creation

---
Issue link: [AIRFLOW-5825](https://issues.apache.org/jira/browse/AIRFLOW-5825)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

potiuk commented on airflow/providers/amazon/aws/operators/sagemaker_endpoint.py:
```diff
@@ -66,6 +68,7 @@ class SageMakerEndpointOperator(SageMakerBaseOperator):
         finish within max_ingestion_time seconds. If you set this parameter to None it never times out.
     :type max_ingestion_time: int
     :param operation: Whether to create an endpoint or update an endpoint. Must be either 'create or 'update'.
+        if self.operation not in ['create', 'update']:
```
I think that's here by mistake :)

potiuk commented on airflow/providers/amazon/aws/operators/sagemaker_endpoint.py:
```diff
@@ -120,6 +123,7 @@ def execute(self, context):
             self.hook.create_endpoint_config(endpoint_config_info)
 
         if self.operation == 'create':
+
```
No need for that empty line?

potiuk reviewed (commented):
Looks good just two small nits/

OmairK committed:
[AIRFLOW-5825] SageMakerEndpointOperator is not idempotent
	- Added try except statement to check for duplicate endpoint while executing a create_endpoint request which if found the request  changes to update_endpoint
	- Added test case to mock the duplicate endpoint creation
